azurelinuxagent/common/cgroupstelemetry.py

```diff
@@ -34,15 +35,24 @@ def _get_metrics_list(metric):
     @staticmethod
     def _process_cgroup_metric(cgroup_metrics):
         current_memory_usage, max_memory_levels, current_cpu_usage = cgroup_metrics.get_metrics()
-        processed_extension = {
-            "memory": {
-                "cur_mem": CGroupsTelemetry._get_metrics_list(current_memory_usage),
-                "max_mem": CGroupsTelemetry._get_metrics_list(max_memory_levels)
-            },
-            "cpu": {
-                "cur_cpu": CGroupsTelemetry._get_metrics_list(current_cpu_usage)
-            }
-        }
+
+        processed_extension = {}
+
+        if current_cpu_usage.count() > 0:
```

Comment:
the consumer of this object will have to deal with the fact that sometimes some of these values will not be there... is that ok?

Reply:
The `cgroup_metrics.get_metrics()` call returns 3 objects of type `Metric`. The list:data inside the Metric would capture all the details of not being able to get data sometimes. If we were not able to capture any data, the length in that case would be 0. Thus the check.

Reply:
let's sync offline, please ping me